### PR TITLE
bug fix: echo now prints entire message on one line

### PR DESCRIPTION
**Pull request by nuclearfall**

### CODE REVIEW

**Comment by vygr** [comment]
Is that a bug ? I'm not sure if it is or not ? If we are folowing Linux style 'echo' then ok it's a fix.

File: cmd/echo.lisp [hunk 1]
```diff
@@ -11,4 +11,4 @@
 
 ;initialize pipe details and command args, abort on error
 (when (and (defq slave (create-slave)) (defq args (options slave usage)))
-	(each print (slice 1 -1 args)))
+	(print (apply cat (map (lambda (_) (cat _ " ")) (slice 1 -1 args)))))
```
Comment (vygr) [inline cmd/echo.lisp]: Won't this have an extra space added on the line that goes to stdout ? ;)

**Comment by vygr** [comment]
(apply cat (map (lambda (_) (cat _ " "))

A little know fact might help you here: during an (each) or (some) or (map) the _ symbol is bound to the element index ! You can do this to avoid the space.

(apply cat (map (lambda (x) (if (= 0 _) x (cat " " x)))))

ie only add the space to the first element and at the front of the others.

or you can just do:

(trim-end (apply cat (map (lambda (_) (cat _ " ")))

Chris

File: cmd/echo.lisp (hunk 1 above)
Comment (nuclearfall) [reply cmd/echo.lisp]: LOL. Oops.

**Comment by vygr** [comment]
Just added (join) to boot.inc as a general case !

Takes a list of sequences and a join sequence and returns a sequence of everything joined. 

As trim only works with char sequences I thought providing this general function was a good idea. 

So you can now just (join (slice 1 -1 args) “ “)

**Pull request closed without merging**